### PR TITLE
MM-38624 Migrate ChannelMemberHistoryStore to sqlx

### PR DESCRIPTION
**Pull request by pikami**
#### Summary
Migrate ChannelMemberHistoryStore to sqlx

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/18408

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
It is failing with `missing destination name email in *[]*model.ChannelMemberHistoryResult` only for Postgres. We do have the `db` struct tag 

```
UserEmail    string `db:"Email"`
```

in the `ChannelMemberHistoryResult` struct.

@pikami - Could you investigate further into it?

**Comment by pikami** [comment]
Thanks for the info @agnivade I'll check it out tomorrow

**Comment by pikami** [comment]
> It is failing with `missing destination name email in *[]*model.ChannelMemberHistoryResult` only for Postgres. We do have the `db` struct tag
> 
> ```
> UserEmail    string `db:"Email"`
> ```
> 
> in the `ChannelMemberHistoryResult` struct.
> 
> @pikami - Could you investigate further into it?

Looks like the email column that is being returned from PostgreSQL is in lowercase ('email') and in MySQL the first letter is uppercase ("Email")
I've fixed this by adding `AS "Email"` statement so it would return the same column name for both DB implementations.

File: store/sqlstore/channel_member_history_store.go [hunk 1]
```diff
@@ -154,7 +157,7 @@ func (s SqlChannelMemberHistoryStore) getFromChannelMembersTable(startTime int64
 	}
 
 	var histories []*model.ChannelMemberHistoryResult
```
Comment (agnivade) [inline store/sqlstore/channel_member_history_store.go:159]: This needs to be `histories := []*model.ChannelMemberHistoryResult{}`

File: store/sqlstore/channel_member_history_store.go [hunk 2]
```diff
@@ -134,7 +137,7 @@ func (s SqlChannelMemberHistoryStore) getFromChannelMemberHistoryTable(startTime
 		return nil, errors.Wrap(err, "channel_member_history_to_sql")
 	}
 	var histories []*model.ChannelMemberHistoryResult
```
Comment (agnivade) [inline store/sqlstore/channel_member_history_store.go:139]: Same as below.

**Review by agnivade — COMMENTED** [review]
Looking good @pikami ! Just some minor things to take care of.

**Comment by agnivade** [comment]
Hey @pikami - checking to see how is it going with this. Just some small things to take care of and it should be good to go.

**Pull request merged**